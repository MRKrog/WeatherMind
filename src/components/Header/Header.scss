.Header {
  width: 100%;
  display: flex;
  justify-content: space-between;
  padding: 0 1em;
  background:#231f20;
}
