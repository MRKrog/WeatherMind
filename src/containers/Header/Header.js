import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import moment from 'moment';

export class Header extends Component {

    render() {
      let time = moment().format('HH:mm:ss');
      return (
        <div className="Header">
          <section className="Header-Time">
            <h4>{time}</h4>
          </section>
          <section className="Header-Search">
            <Route exact path="/" render={() => {
              return <NavLink to="/Search"><i className="fas fa-search"></i></NavLink>
            }} />
            <Route exact path="/Search" render={() => {
              return <NavLink to="/"><i className="fas fa-times-circle"></i></NavLink>
            }} />
          </section>
        </div>
      )
    }
}
