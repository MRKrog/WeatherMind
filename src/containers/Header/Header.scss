.Header {
  width: 100%;
  height: 40px;
  display: flex;
  justify-content: space-between;
  padding: .5em 1.9em .5em 1.25em;
  align-items: baseline;

  .Header-Time {
    line-height: 1;
    h4 {
      margin: 0;
      color: #fff;
      font-weight: 600;
      font-size: .8em;
    }
  }

  .Header-Search {

    a {
      transition: all .5s ease;
      color: #fff;

      &:hover {
        color: #cccccccc;
      }

      i {
        font-size: 1.25em;
      }

    }
  }

}
