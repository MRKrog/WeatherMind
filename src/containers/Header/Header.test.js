import React, { Component } from 'react';
import { Header } from './Header';
import { shallow } from 'enzyme';

jest.mock('moment', () => () => ({format: () => '14:26:37'}));

describe('Header', () => {
  
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

})
