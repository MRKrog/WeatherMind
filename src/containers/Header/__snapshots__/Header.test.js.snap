// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<div
  className="Header"
>
  <section
    className="Header-Time"
  >
    <h4>
      14:26:37
    </h4>
  </section>
  <section
    className="Header-Search"
  >
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Search"
      render={[Function]}
    />
  </section>
</div>
`;
